### PR TITLE
chore: add e2e test for byo public IP

### LINKED ISSUE #618: test: enhance BYO public IP e2e test
When validating BYO public IP for LoadBalancer service, the following steps should be validated:

- create PIP with tags
- create service
- wait for service ready
- verify PIP connectivity
- delete service
- verify PIP still exist
- verify PIP tags don't change
- cleanup

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

chore: add e2e test for byo public IP

**Which issue(s) this PR fixes**:

Fixes #618

**Special notes for your reviewer**:

**Release note**:
```
chore: add e2e test for byo public IP
```

### CODE REVIEW

File: tests/e2e/network/ensureloadbalancer.go [hunk 1]
```diff
@@ -87,6 +87,46 @@ var _ = Describe("Ensure LoadBalancer", func() {
...
+		}()
+
+		By("creating a service referencing the public IP")
+		service := utils.CreateLoadBalancerServiceManifest(testServiceName, nil, labels, ns.Name, ports)
+		service = updateServiceBalanceIP(service, false, targetIP)
+		_, err = cs.CoreV1().Services(ns.Name).Create(context.TODO(), service, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		err = utils.WaitUpdateServiceExposure(cs, ns.Name, testServiceName, targetIP, true)
+		Expect(err).NotTo(HaveOccurred())
+
```
Comment (feiskyer) [inline tests/e2e/network/ensureloadbalancer.go:117]: also check the connectivity of service PIP here?

Comment (nilo19) [reply tests/e2e/network/ensureloadbalancer.go:119]: done

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**